src/input_xml.F90

```diff
@@ -5628,17 +4632,206 @@ subroutine generate_rpn(cell_id, tokens, output)
+          else
+            if (.not. check_for_node(node_nuc, "ao") .and. &
+                 .not. check_for_node(node_nuc, "wo")) then
+              call fatal_error("No atom or weight percent specified for &
+                   &nuclide" // trim(name))
+            elseif (check_for_node(node_nuc, "ao") .and. &
+                    check_for_node(node_nuc, "wo")) then
+              call fatal_error("Cannot specify both atom and weight percents &
+                   &for a nuclide: " // trim(name))
+            end if
+
+            ! Copy atom/weight percents
+            if (check_for_node(node_nuc, "ao")) then
+              call get_node_value(node_nuc, "ao", temp_dble)
+              call densities % push_back(temp_dble)
+            else
+              call get_node_value(node_nuc, "wo", temp_dble)
+              call densities % push_back(-temp_dble)
+            end if
+          end if
+        end do INDIVIDUAL_NUCLIDES
+      end if
+
+      ! ========================================================================
+      ! SET MATERIAL ATOM DENSITIES
+
+      ALL_NUCLIDES: do j = 1, mat % n_nuclides
+        mat % atom_density(j) = densities % data(j)
+      end do ALL_NUCLIDES
+
+      ! Check to make sure either all atom percents or all weight percents are
+      ! given
+      if (.not. (all(mat % atom_density >= ZERO) .or. &
+           all(mat % atom_density <= ZERO))) then
+        call fatal_error("Cannot mix atom and weight percents in material " &
+             // to_str(mat % id))
+      end if
+
+      ! Determine density if it is a sum value
+      if (sum_density) mat % density = sum(mat % atom_density)
+
+      ! Clear lists
+      call densities % clear()
+    end do
+
+  end subroutine assign_nuclide_densities
+
+!===============================================================================
+! NORMALIZE_AO Normalize the nuclide atom percents
 !===============================================================================
 
   subroutine normalize_ao()
-    integer        :: i               ! index in materials array
-    integer        :: j               ! index over nuclides in material
-    real(8)        :: sum_percent     ! summation
-    real(8)        :: awr             ! atomic weight ratio
-    real(8)        :: x               ! atom percent
-    logical        :: percent_in_atom ! nuclides specified in atom percent?
-    logical        :: density_in_atom ! density specified in atom/b-cm?
+    integer                 :: i               ! index in materials array
+    integer                 :: j               ! index over nuclides in material
+    real(8)                 :: sum_percent     ! summation
+    real(8)                 :: awr             ! atomic weight ratio
+    real(8)                 :: x               ! atom percent
+    logical                 :: percent_in_atom ! nuclides specified in atom percent?
+    logical                 :: density_in_atom ! density specified in atom/b-cm?
```

Comment:
I added some other variables (with long initializers) in there in a previous commit and then took them out without readjusting the spaces. I'll remove the extra spaces.